processing/src/main/java/io/druid/jackson/DefaultObjectMapper.java

```diff
@@ -53,6 +53,7 @@ public DefaultObjectMapper(JsonFactory factory)
     configure(MapperFeature.AUTO_DETECT_FIELDS, false);
     configure(MapperFeature.AUTO_DETECT_IS_GETTERS, false);
     configure(MapperFeature.AUTO_DETECT_SETTERS, false);
+    configure(MapperFeature.ALLOW_FINAL_FIELDS_AS_MUTATORS, false);
     configure(SerializationFeature.INDENT_OUTPUT, false);
```

Comment:
this solves some but not actually all the problem, say the field "rowFlushBoundary" in HadoopTuningConfig wasn't a final field, then the test will still pass which is again weird (it happens because jackson auto detected the field)
however, i believe, we should add this as it takes away at least some unintended magic if not all.